### ISSUE TITLE
[receiver/hostmetricsreceiver] Test failure in internal/scraper/processscraper

### ISSUE THREAD

**Issue by jade-guiton-dd** [open-telemetry/opentelemetry-collector-contrib#36468]
### Component(s)

receiver/hostmetrics

## Description

When running the tests for `receiver/hostmetricsreceiver` locally, I consistently get a failure in `receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go` on lines 179 and 180:
```go
internal.AssertContainsAttribute(t, attr, conventions.AttributeProcessCommand)
internal.AssertContainsAttribute(t, attr, conventions.AttributeProcessCommandLine)
```

## Steps to Reproduce

Run `go test` in `receiver/hostmetricsreceiver`.

## Expected Result

Tests pass.

## Actual Result

Tests fail.

### Collector version

Latest main (75ceb559)

## Environment
OS: macOS Sonoma 14.7.1 (on M3 hardware)
Compiler: go 1.22.9

### Log output

```shell
Error:          Should be true
Test:           TestScrape/Enable_memory_utilization
testutils.go:17: 
Error Trace:    [...]/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/testutils.go:17
                [...]/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:179
                [...]/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:128
Error:          Should be true
Test:           TestScrape/Enable_memory_utilization
testutils.go:17: 
Error Trace:    [...]/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/testutils.go:17
                [...]/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:180
                [...]/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:128
Error:          Should be true
Test:           TestScrape/Enable_memory_utilization

=== FAIL: internal/scraper/processscraper TestScrape (re-run 1) (0.25s)
```

**Comment by VihasMakwana** [comment]
@jade-guiton-dd can you run the tests as sudo and see if they work? 
They're failing if you run with a "normal" user, most probably due to insufficient permissions.

**Comment by jade-guiton-dd** [comment]
@VihasMakwana You're right, running the tests as root works. Is this expected behavior? If yes, it may be nice to document it, and/or change test expectations depending on the current user, since I doubt most devs would run local tests as root by default.

**Comment by braydonk** [comment]
Just took a look at this. The test does pass for me locally on a Linux machine, but in reading the test code I found an important mistake that makes it a generally broken test.

I don't have a Mac and know very little about darwin in general, but I can at least point out where the permissions issue is. Getting the command/command line of the process happens here: 

https://github.com/shirou/gopsutil/blob/62a52f910bd9ad7d48a7416f5c2899386d50cb37/process/process_darwin.go#L364   

My guess is that this syscall requires the same permissions as the owner of the process to read this information. (Operating word being "guess" there). If that is the case, then the reason this fails on Mac is that the first process is owned by `root` (so probably pid 1 or something along those lines) and it can't get the command line for it, causing the test to fail.

However, as I mentioned this assertion is busted. While looking at this myself I caught this:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4144b2123f57c34aec4a05a6a5231dad27b5f672/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go#L174-L175

It's just checking the first process over and over again. The fact that it works locally on Linux is actually just because of this issue. When I fix this to look at every resource in the  scraped metrics, it fails basically every test on some random attribute being missing. I'm going to try and come up with a new way to do this test that isn't an assertion of every resource attribute being present; we have a few known errors of special processes missing particular attributes in a way that is completely fine and expected, and this test when actually running like it should will fail on those. I'll probably make it check for only a representative subset or something, I have to mess with it a bit.
